### ISSUE TITLE
no-useless-escape doesn't report unnecessary escaping in character classes

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7424]
**Tell us about your environment**

* **ESLint Version:** 3.8.1
* **Node Version:** 6.9.1
* **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

(none)

**What did you do? Please include the actual source code causing the issue.**

```bash
echo '/[\?]/' | eslint --stdin --no-eslintrc --rule 'no-useless-escape: error'
```

**What did you expect to happen?**

`no-useless-escape` should report that the `?` in the regular expression was uselessly escaped, because `?` has no special meaning in a character class.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors were reported.

A similar issue applies for characters like `-`, although it's a bit trickier because `-` is only sometimes a special character in a regex.

```js
/\-/; // useless escape
/[a\-]/; // useless escape
/[a\-b]/; // not a useless escape
```

`no-useless-escape` does not report useless escaping for any of the above examples.
